addons/account_peppol/models/account_move_send.py

```diff
@@ -13,25 +13,58 @@ class AccountMoveSend(models.AbstractModel):
 
     def _get_alerts(self, moves, moves_data):
         # EXTENDS 'account'
+        def peppol_partner(moves):
+            return moves.partner_id.commercial_partner_id
+
+        def filter_peppol_state(moves, state):
+            return peppol_partner(moves.filtered(
+                lambda m: peppol_partner(m).with_company(m.company_id).peppol_verification_state == state))
+
         alerts = super()._get_alerts(moves, moves_data)
-        if peppol_moves := moves.filtered(lambda m: 'peppol' in moves_data[m]['sending_methods']):
-            invalid_partners = peppol_moves.filtered(
-                lambda move: move.partner_id.commercial_partner_id.with_company(move.company_id).peppol_verification_state != 'valid'
-            ).partner_id.commercial_partner_id
-            ubl_warning_already_displayed = 'account_edi_ubl_cii_configure_partner' in alerts
-            if invalid_partners and not ubl_warning_already_displayed:
-                alerts['account_peppol_warning_partner'] = {
-                    'message': _("The following partners are not correctly configured to receive Peppol documents. "
-                                 "Please check and verify their Peppol endpoint and the Electronic Invoicing format"),
-                    'action_text': _("View Partner(s)"),
-                    'action': invalid_partners._get_records_action(name=_("Check Partner(s)")),
-                }
-            edi_modes = [move.company_id.account_edi_proxy_client_ids.filtered(lambda usr: usr.proxy_type == 'peppol').edi_mode for move in peppol_moves]
-            if any(edi_mode in ('test', 'demo') for edi_mode in edi_modes):
+        # Check for invalid peppol partners.
+        peppol_moves = moves.filtered(lambda m: 'peppol' in moves_data[m]['sending_methods'])
+        if (
+                (invalid_partners := filter_peppol_state(peppol_moves, 'not_valid_format'))
+                                    and not 'account_edi_ubl_cii_configure_partner' in alerts
+        ):
```

Comment:
Cosmetic comment: I don't like the `and` in the middle of the condition :nauseated_face: